lib/signatures.ts

```diff
@@ -12,31 +12,23 @@ export async function getSignaturesByAlgorithm(
   filePaths: string[],
   hashType: SignatureHashAlgorithm = 'dubhash',
 ): Promise<(SignatureResult | null)[]> {
-  if (!isHashValidType(hashType)) {
+  if (hashType !== 'dubhash' && hashType !== 'uhash') {
     throw new Error(`Unsupported hashType ${hashType}`);
   }
 
-  const signatureMapper = getSignatureMapper(baseDir, hashType);
-
-  return await pMap(filePaths, signatureMapper, {
-    concurrency: 20,
-  });
-}
-
-function isHashValidType(hashType: string): boolean {
-  return hashType === 'dubhash' || hashType === 'uhash';
-}
+  const getSignature =
+    hashType === 'uhash' ? getUhashSignature : getDubHashSignature;
 
-const getSignatureMapper = (baseDir: string, hashType: string) => async (
-  filePath: string,
-) => {
-  const absolutePath = join(baseDir, filePath);
-  const fileContents = await readFile(absolutePath);
+  const mapper = async (path: string) => {
```

Comment:
I would suggest to avoid closure functions inside of another functions, it's hard to test and creates coupling.

Would be nice to keep mapper as it was

from
`const mapper = async (path: string) =>`
to
`function getSignatureMapper`

easier to test and make sure that we are using the expected mapper.